components/scream/CMakeLists.txt

```diff
@@ -127,7 +127,7 @@ endif ()
 
 # Add CUDA as a language for CUDA builds
 if (CUDA_BUILD)
-    enable_language(CUDA)
+  enable_language(CUDA)
```

Comment:
Not for this PR, but... we don't actually use CUDA as a cmake language. That is, we don't have any source file, for which we tell CMake that the language is CUDA, and that CMAKE_CUDA_COMPILER needs to be used to build it...

Reply:
Imho, it would be a nice EKAT issue, to set compilers "the right way". That means, CMAKE_CXX_COMPILER=mpicxx, CMAKE_CUDA_COMPILER=nvcc. But I fear that the solution would be "set LANG=CUDA for all cxx files on GPU builds", since virtually all cxx files eventually include some kokkos header...

Reply:
Being very cmake-style, we should actually set `CMAKE_CXX_COMPILER=g++`, and then call `find_package(MPI)`, and link against MPI.

Reply:
Sorry for the long thread, I just had a stream of thoughts. I feel like all of these changes to make scream "very cmake compliant" should be tried in EKAT at some point, to experiment.